run-tests.sh

```diff
@@ -1,17 +1,73 @@
-#!/bin/sh
+#!/bin/bash
 #
 # This file is part of REANA.
-# Copyright (C) 2017, 2018 CERN.
+# Copyright (C) 2017, 2018, 2019, 2020 CERN.
 #
 # REANA is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
-pydocstyle reana_server && \
-black --check . && \
-FLASK_APP=reana_server/app.py python ./scripts/generate_openapi_spec.py && \
-diff -q -w temp_openapi.json docs/openapi.json && \
-check-manifest --ignore ".travis-*" && \
-sphinx-build -qnN docs docs/_build/html && \
-REANA_SQLALCHEMY_DATABASE_URI=sqlite:// python setup.py test && \
-sphinx-build -qnN -b doctest docs docs/_build/html && \
+# Quit on errors
+set -o errexit
+
+# Quit on unbound symbols
+set -o nounset
+
+export REANA_SQLALCHEMY_DATABASE_URI=postgresql+psycopg2://postgres:mysecretpassword@localhost/postgres
+
+# Verify that db container is running before continuing
+_check_ready() {
+    RETRIES=40
+    while ! $2
+    do
+        echo "==> [INFO] Waiting for $1, $((RETRIES--)) remaining attempts..."
+        sleep 2
+        if [ $RETRIES -eq 0 ]
+        then
+            echo "==> [ERROR] Couldn't reach $1"
+            exit 1
+        fi
+    done
+}
+
+_db_check() {
+    docker exec --user postgres postgres__reana-server bash -c "pg_isready" &>/dev/null;
+}
+
+clean_old_db_container() {
+    OLD="$(docker ps --all --quiet --filter=name=postgres__reana-server)"
+    if [ -n "$OLD" ]; then
+        echo '==> [INFO] Cleaning old DB container...'
+        docker stop postgres__reana-server
+    fi
+}
+
+start_db_container() {
+    echo '==> [INFO] Starting DB container...'
+    docker run --rm --name postgres__reana-server -p 5432:5432 -e POSTGRES_PASSWORD=mysecretpassword -d postgres:9.6.2
+    _check_ready "Postgres" _db_check
+}
+
+stop_db_container() {
+    echo '==> [INFO] Stopping DB container...'
+    docker stop postgres__reana-server
+}
+
+check_black() {
+    echo '==> [INFO] Checking Black compliance...'
+    black --check .
+}
+
+pydocstyle reana_server
+check_black
+FLASK_APP=reana_server/app.py python ./scripts/generate_openapi_spec.py
+diff -q -w temp_openapi.json docs/openapi.json
+
```

Comment:
In place of empty newline, add `rm temp_openapi.json` statement as in `reana-workflow-controller`.